### ISSUE TITLE
Conditional loading on config does not use defaultExtension

### ISSUE THREAD

**Issue by filipesilva** [systemjs/systemjs#1269]
Trying to conditionally load a file without extension within `System.config` (but with `defaultExtension` defined for the package) doesn't seem to work. Adding the extension manually works fine though.

```
  packages: {
    'env':{
      defaultExtension: 'ts',
      "map": {
        "./environment": "./environment.#{env/current-environment}"
      }
    }
  },
```

Plunker: http://plnkr.co/edit/0LXFkQP60HfJWeAkgMUy?p=preview

**Comment by guybedford** [comment]
Sure we can add support for this.

**Comment by filipesilva** [comment]
Cool, thank you :D

**Comment by guybedford** [comment]
Fixed in https://github.com/systemjs/systemjs/commit/7c673602d6eeabe171f941f4579e00d4f462e985.

**Comment by filipesilva** [comment]
Nice, looking forward to it!

**Comment by guybedford** [comment]
Released in 0.19.28.
